tracking/src/SessionManager.js

```diff
@@ -80,24 +70,22 @@ class SessionManager {
     }
 
     
+    
     getSessionId() {
         let now = new Date().getTime();
-        if (now - this.sessionTimestamp > SESSION_TIMEOUT || !this.isSameDay(now, this.sessionTimestamp)) {
+        let sid = this.getLocalStorage(KEY_SID); 
+        if (!sid || (now - this.sessionTimestamp) > SESSION_TIMEOUT || !this.isSameDay(now, this.sessionTimestamp)) {
             this.sessionTimestamp = now; 
-            this.sid = this.generateUUID(); 
-            if(typeof(Storage) !== 'undefined') {
-                localStorage.setItem(KEY_SID, this.sid); 
-                localStorage.setItem(KEY_SID_TIMESTAMP, this.sessionTimestamp); 
-            }
+            sid = this.generateUUID(); 
+            this.setLocalStorage(KEY_SID, sid); 
+            this.setLocalStorage(KEY_SID_TIMESTAMP, this.sessionTimestamp); 
+
         } else {
             this.sessionTimestamp = now; 
```

Comment:
should be storing in local storage instead of memory

Reply:
changed already